datatypes/src/operations/reproject.rs

```diff
@@ -218,8 +218,111 @@ where
+    bbox: BoundingBox2D,
+    spatial_resolution: SpatialResolution,
+    projector: &P,
+) -> Result<SpatialResolution> {
+    // calculate the number of pixels per axis
+    let x_pixels = (bbox.size_x() / spatial_resolution.x).abs();
+    let y_pixels = (bbox.size_y() / spatial_resolution.y).abs();
+    // get the diagonal distance between bbox edges
+    let diag_pixels: f64 = (x_pixels * x_pixels + y_pixels * y_pixels).sqrt();
+
+    // reproject the upper left and lower right coordinates to the target srs.
+    // NOTE: the edges of the projected bbox might differ.
+    let proj_ul_coord = bbox.upper_left().reproject(projector)?;
+    let proj_lr_coord = bbox.lower_right().reproject(projector)?;
+
+    // calculate the distance between upper left and lower right coordinate in srs units
+    let proj_ul_lr_vector = proj_ul_coord - proj_lr_coord;
+    let proj_ul_lr_distance = (proj_ul_lr_vector.x * proj_ul_lr_vector.x
+        + proj_ul_lr_vector.y * proj_ul_lr_vector.y)
+        .sqrt();
+
+    // derive the pixel size by deviding srs unit distance by pixel distance in the source bbox
+    let proj_ul_lr_pixel_size = proj_ul_lr_distance / diag_pixels;
+    Ok(SpatialResolution::new_unchecked(
+        proj_ul_lr_pixel_size,
+        proj_ul_lr_pixel_size,
+    ))
+}
+
+/// This approach uses the GDAL way to suggest the pixel size. However, we check both diagonals and take the smaller one.
+pub fn suggest_pixel_size_from_diag_cross<P: CoordinateProjection>(
+    bbox: BoundingBox2D,
+    spatial_resolution: SpatialResolution,
+    projector: &P,
+) -> Result<SpatialResolution> {
+    // calculate the number of pixels per axis
+    let x_pixels = (bbox.size_x() / spatial_resolution.x).abs();
+    let y_pixels = (bbox.size_y() / spatial_resolution.y).abs();
+
+    let diag_pixels: f64 = (x_pixels * x_pixels + y_pixels * y_pixels).sqrt();
+
+    let proj_ul_coord = bbox.upper_left().reproject(projector)?;
+    let proj_lr_coord = bbox.lower_right().reproject(projector)?;
+
+    let proj_ul_lr_vector = proj_ul_coord - proj_lr_coord;
+    let proj_ul_lr_distance = (proj_ul_lr_vector.x * proj_ul_lr_vector.x
+        + proj_ul_lr_vector.y * proj_ul_lr_vector.y)
+        .sqrt();
+
+    let proj_ul_lr_pixel_size = proj_ul_lr_distance / diag_pixels;
+
+    let proj_ll_coord = bbox.lower_left().reproject(projector)?;
+    let proj_ur_coord = bbox.upper_right().reproject(projector)?;
+
+    let proj_ll_ur_vector = proj_ll_coord - proj_ur_coord;
+    let proj_ll_ur_distance = (proj_ll_ur_vector.x * proj_ll_ur_vector.x
+        + proj_ll_ur_vector.y * proj_ll_ur_vector.y)
+        .sqrt();
+
+    let proj_ll_ur_pixel_size = proj_ll_ur_distance / diag_pixels;
+
+    let min_pixel_size = proj_ll_ur_pixel_size.min(proj_ul_lr_pixel_size);
+
+    Ok(SpatialResolution::new_unchecked(
+        min_pixel_size,
+        min_pixel_size,
+    ))
+}
+
+pub fn project_coordinates_fail_tolarant<P: CoordinateProjection>(
```

Comment:
Maybe add a comment what happens here. My guess:

```suggestion
/// Tries to reproject all coordinates at once. If this fails, tries to reproject coordinate by coordinate.
/// If this still fails, it returns some for individual coordinates.
/// Otherwise, returns a vector of `Some(Coordinate2d)`.
pub fn project_coordinates_fail_tolarant<P: CoordinateProjection>(
```